c3.js

```diff
@@ -7445,8 +7450,12 @@ c3_chart_internal_fn.updateLegend = function (targetIds, options, transitions) {
         background = $$.legend.insert('g', '.' + CLASS.legendItem).attr("class", CLASS.legendBackground).append('rect');
     }
 
-    texts = $$.legend.selectAll('text').data(targetIds).text(function (id) {
-        return isDefined(config.data_names[id]) ? config.data_names[id] : id;
+    texts = $$.legend.selectAll('text').data(targetIds).text(function (id, index) {
+        if(config.legend_format && isFunction(config.legend_format)) {
+            return config.legend_format(id, index);
+        } else {
+            return isDefined(config.data_names[id]) ? config.data_names[id] : id;
```

Comment:
Please, do not repeat the logic that resolve the legend's name.

Reply:
I didn't get you here.

Reply:
@pkpatels I saw the same logic twice but it must have been between c3.js and legend.js files.

Your pull request still includes the `c3.js` file although it should not be committed.